ypathgen/pathgen.go

```diff
@@ -62,6 +63,9 @@ const (
 	// NOTE: This cannot be "", as the builder method name would conflict
 	// with the child constructor method for the keys.
 	BuilderKeyPrefix = "With"
+	// defaultFakeRootPkgName is the default name for the package that contains
+	// the fake root struct (if splitting by module is enabled).
+	defaultFakeRootPkgName = "device"
```

Comment:
Sorry for not catching this earlier. I'm thinking instead of putting the fake root inside its own package, that we should still write the fake root in `ocPathStructsOutputFile`, which is the same file we'd be writing in if we didn't split packages. We should also make sure `generatePathStructsMultipleFiles` in generator.go is false so that we're guaranteed `ocPathStructsOutputFile` exists.

Reply:
i think that may be a bit confusing because if split by modules is true then you need to specify both output_dir and path_struct_output_file. Also fake root generation is optional. so `ocPathStructsOutputFile` needs be optional as well. The fake root is already its own package. What's the goal of writing the fake root `ocPathStructsOutputFile`?

Reply:
I agree it makes the flags even more complicated. However, I'm thinking there is value in separating out the API that's accessible to users (i.e. the fake root's package), and the API that's not accessible. I believe fakeroot generation for PathStructs is **mandatory**, indeed `ygot.ResolvePath` relies on this fact. I don't see how users would use PathStructs without the fakeroot, so I think it's reasonable to assume it'll always be there.

Therefore, I think we need to check that if split-by-modules is true, that both flags are set. What do you think?

Reply:
sgtm, added checks and update flag docs.